image/cas/interface.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cas implements generic content-addressable storage.
+package cas
+
+import (
+	"io"
+
+	"golang.org/x/net/context"
+)
+
+// Engine represents a content-addressable storage engine.
+//
+// This interface is for internal use of oci-image-tool for the time
+// being.  It is subject to change.  This notice will be removed when
+// and if the interface becomes stable.
+type Engine interface {
```

Comment:
On Tue, Aug 30, 2016 at 01:38:05PM -0700, Stephen Day wrote:
> That is not at all what I said. I am not suggesting a new
> method. `Copy` would be a generic implementation over the `Engine`
> interface.

Ok, so exactly the pattern I already use in image/cas/put.go for
PutJSON.  Would you like me to add image/cas/copy.go with your:

  Copy(dstEngine, srcEngine Engine, digest string) error

?